### PR TITLE
DR2-2007 Remove mgmt terraform roles and use account ones.

### PR DESCRIPTION
**Pull request by MancunianSam**
This deletes the management roles which assumed the roles in the
environment accounts and updates the trust policies on those accounts to
allow GitHub to run them.

It also updates the bucket policy for the state bucket and the resource
policy for the lock table.

This will need https://github.com/nationalarchives/da-terraform-modules/pull/82 to be merged 
before this can be deployed.

### CODE REVIEW

File: templates/iam_policy/terraform_policy.json.tpl [hunk 1]
```diff
@@ -55,6 +55,23 @@
...
+        "dynamodb:DeleteItem",
+        "dynamodb:GetItem",
+        "dynamodb:PutItem",
+        "s3:GetObject",
+        "s3:ListBucket",
+        "s3:PutObject"
+      ],
+      "Effect": "Allow",
+      "Resource": [
+        "arn:aws:s3:::mgmt-dp-terraform-state",
```
Comment (techncl) [inline templates/iam_policy/terraform_policy.json.tpl:71]: Is it fine that we're hardcoding `mgmt` here and below?

File: root.tf [hunk 2]
```diff
@@ -158,49 +111,53 @@ module "environment_roles_intg" {
...
     aws = aws.intg
   }
-  source                        = "./environment_roles"
-  account_number                = data.aws_ssm_parameter.intg_account_number.value
-  environment                   = "intg"
-  management_account_number     = data.aws_caller_identity.current.account_id
-  terraform_external_id         = module.terraform_config.terraform_config["intg"]["terraform_external_id"]
-  terraform_github_role_arn     = module.terraform_github_terraform_environments["intg"].role_arn
-  management_developer_role_arn = data.aws_ssm_parameter.dev_admin_role.value
+  source                    = "./environment_roles"
```
Comment (techncl) [inline root.tf:114]: Should the source be adding to a local variable since it's used 4 times?

**Review by techncl — COMMENTED** [review]
Looks good! Just 2 minor comments

File: templates/iam_policy/terraform_policy.json.tpl (hunk 1 above)
Comment (MancunianSam) [reply templates/iam_policy/terraform_policy.json.tpl:71]: Yeah, there's only one state bucket and it's in the management account.

File: root.tf (hunk 2 above)
Comment (MancunianSam) [reply root.tf:114]: I just tried it, terraform won't let you. [A lot of people want it](https://github.com/hashicorp/terraform/issues/1439) but they say no.

**Review by techncl — APPROVED** [review]
👍

**Pull request merged**